engine/src/flutter/lib/ui/platform_dispatcher.dart

```diff
@@ -3189,3 +3211,22 @@ enum ViewFocusDirection {
   /// This is typically result of the user pressing shift + tab.
   backward,
 }
+
+/// The request containing all the information required to perform framework hitTest.
+class HitTestRequest {
```

Comment:
@chunhtai If turns out these cannot be `@immutable`. `@immutable` comes from `package:meta`, which `dart:ui` does not have access to.

Reply:
makes sense, I guess we will have to wait for dart support for data class then.